tutorials/haskell/gui-application/tutorial.md

```diff
@@ -0,0 +1,825 @@
+  description). “Direct translation” means that you pass around pointers.
+  Yes, in Haskell code. For documentation authors suggest look
+  [here](https://tronche.com/gui/x/xlib/) (although Haddocks are not blank
+  either). Last release was in May 2014.
+
+* [`hsqml`](https://hackage.haskell.org/package/hsqml) — the Haskell
+  bindings to
+  [Qt Quick](https://wiki.qt.io/Category:Developing_with_Qt::Qt_Quick).
+  Missing from Stackage. Releases seem to happen once a year.
+
+* [`fltkhs`](https://hackage.haskell.org/package/fltkhs) — the Haskell
+  bindings to the [FLTK GUI library](http://www.fltk.org/index.php). The
+  package seems to be pretty new and I personally don't see any benefit in
+  using it over more mature solutions.
+
+As to GTK+, it seems to be:
+
+* Easy to install.
+
+* Robust and mature. Documentation is quite good and comprehensive.
+
+It's worth mentioning that Haskell has developed much stronger ecosystem
+with respect to web-development than GUI and desktop development in general.
+For further reference about state of standalone GUI applications in Haskell,
+refer to:
+
+* The section in Gabriel Gonzalez's
+  [State of Haskell Ecosystem](https://github.com/Gabriel439/post-rfc/blob/master/sotu.md#standalone-gui-applications)
+
+* The [blog post](http://keera.co.uk/blog/2014/05/23/state-gui-programming-haskell/) by Keera Studios
+
+## Installation
+
+Please see [this article](https://wiki.haskell.org/Gtk2Hs/Installation) for
+Gtk2Hs installation. It has instructions for Windows, Linux, Mac, and
+FreeBSD.
+
+## First steps
+
+The calculator application has been chosen because its logic is very
+straightforward and we can focus on working with GTK+ framework without much
+distraction while keeping the tutorial reasonably practical.
+
+Let's start by importing some modules:
+
+```haskell
+import Control.Monad
+import Control.Monad.IO.Class
+import Data.IORef
+import Graphics.UI.Gtk hiding (Action, backspace)
+```
+
+As I said, GTK+ bindings are very imperative. All binding code lives in `IO`
+monad except for some cases that we will cover shortly.
+
+```haskell
+main :: IO ()
+main = do
+  void initGUI          -- (1)
+  window <- windowNew   -- (2)
+                        -- (3)
+  widgetShowAll window  -- (4)
+  mainGUI               -- (5)
+```
+
+1. The first thing we need to do in any program that uses GTK+ is to call
+   the `initGUI` function. This function allocates some resources and
+   prepares GTK+ for work, it also looks up command line arguments that are
+   relevant to GTK+, parses them, and returns all non-parsed arguments. For
+   our purposes, we don't need the command line arguments, so let's wrap it
+   with the `void`.
+
+2. Next, we need a window to build interface of our calculator inside it. To
+   create a new top-level window we use the `newWindow` action. It returns
+   opaque `Window` value that can be used to manipulate the window.
```

Comment:
@mrkkrp 

- "to build **the** interface of our calculator inside it".
- "It returns **an** opaque `Window` value".